### DISCUSSION TITLE
Not able to Communicate in terminal

### DISCUSSION THREAD

**Discussion by jishnuj74** [eProsima/Fast-DDS#3864] [Unofficial support]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

I should able to publish ot list the topic in the another terminal

### Current behavior

In one terminal i kept export ROS_LOCALHOST_ONLY=1 and im publishing a oublisher in that terminal in another terminal export ROS_LOCALHOST_ONLY=0 , but im not able to see the topic of this publisher while ted over networkconnecconnecting over network if i disabled the wifi and test in same scenario im able to this topic and list it in another terminal.

### Steps to reproduce

1. Terminal 1 export ROS_LOCALHOST_ONLY=1 . then type ros2 run demo_node_cpp publisher
2. Terminal 2 export ROS_LOCALHOST_ONLY=0 . type in that terminal ros2 topic list

### Fast DDS version/commit

Fast rtps  DDS version -   2.6.6-1jammy.20230807.150657

 ros-humble-rmw-fastrtps-cpp  - 6.2.3-1jammy.20230807.152615      

### Platform/Architecture

Other. Please specify in Additional context section.

### Transport layer

UDPv4

### Additional context

Ubuntu 22.04  -LTS ros2 humble

### XML configuration file

```xml
Not availabe
```

### Relevant log output

```shell
not availabe
```

### Network traffic capture

not availabe

**Comment by JesusPoderoso** [comment]
Hi @jishnuj74 
I have reproduced your steps and the `/chatter` topic is displayed in the topics list.
Also, this issue is not using any of the Fast DDS environment variables (`ROS_LOCALHOST_ONLY`), so please consider to open this ticket in the [rmw_fastrtps](https://github.com/ros2/rmw_fastrtps)

I am transferring this ticket to the support discussion forum as per [Fast DDS CONTRIBUTING guidelines](https://github.com/eProsima/Fast-DDS/blob/master/CONTRIBUTING.md)

**Reply by jishnuj74** [reply]
[Hi]([url](url)) @JesusPoderoso  Have you connected to wifi and do the steps. Because im not able to see the chatter topic
![Screenshot from 2023-09-20 16-20-54](https://github.com/eProsima/Fast-DDS/assets/32978036/a02668c3-64ec-45be-b026-767b91e247d5)
